backend/s3/s3.go

```diff
@@ -3159,7 +3159,9 @@ func (f *Fs) newV2List(req *s3.ListObjectsV2Input) bucketLister {
 // Do a V2 listing
 func (ls *v2List) List(ctx context.Context) (resp *s3.ListObjectsV2Output, versionIDs []*string, err error) {
 	resp, err = ls.f.c.ListObjectsV2WithContext(ctx, &ls.req)
-	ls.req.ContinuationToken = resp.NextContinuationToken
+	if err == nil {
```

Comment:
Can you re-write as an early return to make it look like the other listing routines - something like

```go
if err != nil {
    return nil, nil, err
}
ls.req.ContinuationToken = resp.NextContinuationToken
return resp, nil, nil
```